crates/core/src/host/wasmer/wasmer_module.rs

```diff
@@ -174,6 +175,37 @@ impl module_host_actor::WasmInstancePre for WasmerModule {
             .map_err(|err| InitializationError::Instantiation(err.into()))?;
 
         let mem = Mem::extract(&instance.exports).unwrap();
+
+        let abi_version = instance
```

Comment:
I would maybe just put some of the (very good!) explanation of the choice to do this at instantiation time in the commit message here as a comment too.

I agree that writing your own increasingly elaborate abstract interpreter to read an int is not worth failing slightly faster.

Reply:
+1 on the comment.